scripts/runscript

```diff
@@ -0,0 +1,169 @@
+#!/bin/sh
```

Comment:
this script is getting so large that I would consider moving to python.

Reply:
Very good point. I was thinking about this also; I primarily wanted to rewrite the script to "clean it up" and thus kept most of the functions and the structure the same, but I feel like maybe I should have a second go at it to see if it can be better structured (or else made into a python script).